server/apis/v1_1/handler.go

```diff
@@ -0,0 +1,844 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1_1
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	metricsversiond "k8s.io/metrics/pkg/client/clientset/versioned"
+	"k8s.io/utils/pointer"
+
+	dfv1 "github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+	"github.com/numaproj/numaflow/pkg/apis/proto/daemon"
+	dfv1versiond "github.com/numaproj/numaflow/pkg/client/clientset/versioned"
+	dfv1clients "github.com/numaproj/numaflow/pkg/client/clientset/versioned/typed/numaflow/v1alpha1"
+	daemonclient "github.com/numaproj/numaflow/pkg/daemon/client"
+	"github.com/numaproj/numaflow/webhook/validator"
+)
+
+// SpecType is used to provide the type of the spec of the resource
+// This is used to parse different types of specs from the request body
+const (
+	SpecTypePipeline = "pipeline"
+	SpecTypeISB      = "isb"
+	SpecTypePatch    = "patch"
+)
+
+type handler struct {
+	kubeClient     kubernetes.Interface
+	metricsClient  *metricsversiond.Clientset
+	numaflowClient dfv1clients.NumaflowV1alpha1Interface
+}
+
+// NewHandler is used to provide a new instance of the handler type
+func NewHandler() (*handler, error) {
+	var restConfig *rest.Config
+	var err error
+	kubeconfig := os.Getenv("KUBECONFIG")
+	if kubeconfig == "" {
+		home, _ := os.UserHomeDir()
+		kubeconfig = home + "/.kube/config"
+		if _, err := os.Stat(kubeconfig); err != nil && os.IsNotExist(err) {
+			kubeconfig = ""
+		}
+	}
+	if kubeconfig != "" {
+		restConfig, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+	} else {
+		restConfig, err = rest.InClusterConfig()
+	}
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get kubeconfig, %w", err)
+	}
+	kubeClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get kubeclient, %w", err)
+	}
+	metricsClient := metricsversiond.NewForConfigOrDie(restConfig)
+	numaflowClient := dfv1versiond.NewForConfigOrDie(restConfig).NumaflowV1alpha1()
+	return &handler{
+		kubeClient:     kubeClient,
+		metricsClient:  metricsClient,
+		numaflowClient: numaflowClient,
+	}, nil
+}
+
+// ListNamespaces is used to provide all the namespaces that have numaflow pipelines running
+func (h *handler) ListNamespaces(c *gin.Context) {
+	namespaces, err := getAllNamespaces(h)
+	if err != nil {
+		h.respondWithError(c, fmt.Sprintf("Failed to fetch all namespaces, %s", err.Error()))
+		return
+	}
+	c.JSON(http.StatusOK, NewNumaflowAPIResponse(nil, namespaces))
```

Comment:
Similar to `respondWithError`, add a function `respondOK(c *gin.Context, data interface{})`?